src/server/server.js

```diff
@@ -77,21 +223,30 @@ function streamAppHostHtml(request, response) {
             }
         }).pipe(response);
     }).done();
-}
+};
 
-function streamSimHostHtml(request, response) {
+/**
+ * @param {object} request
+ * @param {object} response
+ * @private
+ */
+SimulationServer.prototype._streamSimHostHtml = function (request, response) {
     // If we haven't ever prepared, do so before we try to generate sim-host, so we know our list of plugins is up-to-date.
     // Then create sim-host.js (if it is out-of-date) so it is ready when it is requested.
-    prepare.prepare().then(function () {
-        return simFiles.createSimHostJsFile();
-    }).then(function () {
+    var project = this._project;
+
+    project.prepare().then(function () {
```

Comment:
it would be nice if the call to `.then()` and `.done()` were on the next line (here and below). I believe in other places it was indented also, like this:

```javascript
project.prepare()
     .then(function () {
     }.bind(this))
...
     .done()
```